spec/test_app_templates/Gemfile.extra

```diff
@@ -2,4 +2,18 @@
 
 group :development do
   gem 'linkeddata'
+  case ENV['RAILS_VERSION']
+  when /^6\.[01]\./
+    # Ruby 3.0 comes with Psych 3, while Ruby 3.1 comes with Psych 4,
+    # which has a major breaking change (diff 3.3.2 → 4.0.0).
+    # See: https://bugs.ruby-lang.org/issues/17866
+    # See also: https://stackoverflow.com/questions/71191685/visit-psych-nodes-alias-unknown-alias-default-psychbadalias/71192990#71192990
+    gem 'psych', '< 4'
+    # opt into mail 2.8.0.rc1 so we get extra dependencies
+    # Once mail 2.8.0 final is released this will not be required.
+    # https://github.com/mikel/mail/pull/1472
+    gem "mail", ">= 2.8.0.rc1"
+    # See https://stackoverflow.com/questions/70500220/rails-7-ruby-3-1-loaderror-cannot-load-such-file-net-smtp
+    gem 'net-smtp', require: false
+  end
```

Comment:
Actually, using ruby 3.1 and Rails 6.0 isn't a good idea in general for Rails itself -- so let's only add these in for 6.1, **and** can we put in a condition where we only add them in for ruby 3.1 too? Just to restrict this weird stuff to where we know we need it and not have it infect other test cases.

Reply:
Done. I made it an `if`, not a `case`, too.

Reply:
Yup.